src/components/Search/SearchPageHeader/SearchTypeMenuButton.tsx

```diff
@@ -0,0 +1,200 @@
+                            horizontal: CONST.MODAL.ANCHOR_ORIGIN_HORIZONTAL.RIGHT,
+                            vertical: CONST.MODAL.ANCHOR_ORIGIN_VERTICAL.TOP,
+                        }}
+                        disabled={item.pendingAction === CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE}
+                    />
+                ),
+                styles: [styles.textSupporting],
+                isSelected: false,
+                shouldCallAfterModalHide: true,
+            };
+        },
+        [hash, getOverflowMenu, styles.textSupporting, personalDetails, reports, taxRates, allCards],
+    );
+
+    const savedSearchItems = useMemo(() => {
+        if (!savedSearches) {
+            return [];
+        }
+        return Object.entries(savedSearches).map(([key, item], index) => createSavedSearchMenuItem(item, key, index));
+    }, [createSavedSearchMenuItem, savedSearches]);
+
+    const currentSavedSearch = savedSearchItems.find((item) => Number(item.hash) === hash);
+
+    const popoverMenuItems = useMemo(() => {
+        const items = typeMenuItems.map((item, index) => {
+            const isSelected = title ? false : index === activeItemIndex;
+
+            return {
+                text: translate(item.translationPath),
+                onSelected: singleExecution(() => {
+                    clearAllFilters();
+                    Navigation.navigate(item.getRoute(policyID));
+                }),
+                isSelected,
+                icon: item.icon,
+                iconFill: isSelected ? theme.iconSuccessFill : theme.icon,
+                iconRight: Expensicons.Checkmark,
+                shouldShowRightIcon: isSelected,
+                success: isSelected,
+                containerStyle: isSelected ? [{backgroundColor: theme.border}] : undefined,
+                shouldCallAfterModalHide: true,
+            };
+        });
+
+        if (title && !currentSavedSearch) {
+            items.push({
+                text: title,
+                onSelected: closeMenu,
+                isSelected: !currentSavedSearch,
+                icon: Expensicons.Filters,
+                iconFill: theme.iconSuccessFill,
+                success: true,
+                containerStyle: undefined,
+                iconRight: Expensicons.Checkmark,
+                shouldShowRightIcon: false,
+                shouldCallAfterModalHide: true,
+            });
+        }
+
+        return items;
+    }, [typeMenuItems, title, currentSavedSearch, activeItemIndex, translate, singleExecution, theme.iconSuccessFill, theme.icon, theme.border, policyID, closeMenu]);
+
+    const allMenuItems = [];
+    allMenuItems.push(...popoverMenuItems);
+
+    if (savedSearchItems.length > 0) {
+        allMenuItems.push({
+            text: translate('search.savedSearchesMenuItemTitle'),
+            styles: [styles.textSupporting],
+            disabled: true,
+        });
+        allMenuItems.push(...savedSearchItems);
+    }
```

Comment:
can te logic from lines 163 - 172 (so creating `allMenuItems`) live also inside the `useMemo` block from line 124?

Right now we have:
```ts
const popoverMenuItems = useMemo(() => {
[...]
}

allItems = /* modified popoverMenuItems */

```
and then we only use `allMenuItems` in the code - popoverMenuItems are not needed